### DISCUSSION TITLE
Import RAW from osciloscope

### DISCUSSION THREAD

**Discussion by glococo** [bengtmartensson/IrScrutinizer#505] [Q&A] [answered]
Hi,

I want to import a RAW wave in csv from Osciloscope. (1º col time, 2ºcol value/voltage )

Can anyone share a CSV working sample file ? or CSV import definition ?
I cannot find how to setup CSV in the right way.

Thanks in advance
[UNIT_UTD4202C_0000.csv](https://github.com/bengtmartensson/IrScrutinizer/files/13226317/UNIT_UTD4202C_0000.csv)

**Comment by bengtmartensson** [comment]
A raw IR signal/sequence in IrScrutinizer consists of a sequence of interleaving on- and off-periods, on with +, off with -, with the number denoting the how many micro seconds the on/off period lasts. Your file contains equidistant samples. So you have to write a program/script that converts the samples to the duration format.This is not entirely straightforward, but should not be too hard with elementary programming knowledge, in, e.g., Python.

**Reply by glococo** [reply]
Hi Bengt,
Nice to hear your feedback. I know ~~jiu jitsu~~ nodeJs. but it is even easier, with just LO_Calc.
Here is the CSV with the first column in microseconds.
Do it need to be more refined ? How should I import it ?

[Osciloscope_time_in_us.csv](https://github.com/bengtmartensson/IrScrutinizer/files/13249591/Osciloscope_time_in_us.csv)

Thanks a lot for your patience

**Comment by bengtmartensson** [comment, marked as answer]
Unfortunately, you appear to have misunderstood my first message. What you need is the durations, i.e. the time between the transitions (on -> off, off -> on).

**Reply by glococo** [reply]
I see, something like "add test signal"-> Export in "Text"->raw:

test
+9041 -4507 +573 -573 +573 -573 +573 -1694 +573 -1694 +573 -573 +573 -573 +573 -573 +573 -573 +573 -573 +573 -1694 +573 -573 +573 -573 +573 -573 +573 -1694 +573 -573 +573 -573 +573 -573 +573 -573 +573 -573 +573 -1694 +573 -1694 +573 -1694 +573 -573 +573 -573 +573 -1694 +573 -1694 +573 -1694 +573 -573 +573 -573 +573 -573 +573 -1694 +573 -1694 +573 -44293
+9041 -2267 +573 -96193
